src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/EndpointFactory.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Creates <see cref="Endpoint"/>s by name. 
+    /// </summary>
+    public class EndpointFactory
+    {
+        public EndpointFactory(
+            ValidatorFactory validationFactory, 
+            Func<string, Uri> endpointCursorUriFactory, 
+            Func<HttpMessageHandler> messageHandlerFactory, 
+            ILoggerFactory loggerFactory)
+        {
+            _validationFactory = validationFactory;
+            _endpointCursorUriFactory = endpointCursorUriFactory;
+            _messageHandlerFactory = messageHandlerFactory;
+            _loggerFactory = loggerFactory;
+        }
+
+        /// <summary>
+        /// Creates the <see cref="Endpoint"/> for which <see cref="Type.Name"/> is equal to <paramref name="endpointName"/>.
+        /// </summary>
+        public Endpoint Create(string endpointName)
```

Comment:
Maybe at this level, maybe at a higher level -- up to you.

---
In reply to: [111510819](https://github.com/NuGet/NuGet.Services.Metadata/pull/151#discussion_r111510819) [](ancestors = 111510819)